logstash-input-webfetcher/src/main/java/eu/wajja/web/fetcher/controller/URLController.java

```diff
@@ -63,12 +64,19 @@ public Result getURL(String currentUrl, String initialUrl, String chromeDriver)
 			result.setUrl(currentUrl);
 			result.setRootUrl(initialUrl);
 
-			if (code == HttpURLConnection.HTTP_OK) {
+            if (code == HttpURLConnection.HTTP_OK) {
+                /*If pdf, do not use the webdriver, which was only used intitially to handle js*/
+                if (currentUrl.endsWith("pdf")) {

```

Comment:
cant we read the headers ? it must be defined somewhere. Also imagine the filename is filename.PDF

Reply:
Corrected. Left the extension check, made it case-insensitive, and added the header check.